packages/discovery-provider/src/api/v1/access_helpers.py

```diff
@@ -0,0 +1,51 @@
+import logging
+
+from flask_restx.errors import abort
+
+from src.queries.get_managed_users import (
+    GetUserManagersArgs,
+    get_user_managers_with_grants,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def is_active_manager(user_id: int, manager_id: int) -> bool:
+    """
+    Check if a manager is active for a given user.
+
+    Args:
+        user_id (int): The ID of the user.
+        manager_id (int): The ID of the manager.
+
+    Returns:
+        bool: True if the manager is active for the user, False otherwise.
+    """
+    try:
+        grants = get_user_managers_with_grants(
+            GetUserManagersArgs(user_id=user_id, is_approved=True, is_revoked=False)
+        )
+        for grant in grants:
+            manager = grant.get("manager")
+            if manager and manager.get("user_id") == manager_id:
+                return True
+    except Exception as e:
+        logger.error(
+            f"access_helpers.py | Unexpected exception checking managers for user: {e}"
+        )
+    return False
+
+
+def require_owner_or_manager(user_id, authed_user_id):
```

Comment:
`check_authorized` makes the most sense to me. We really do treat this as an ownership scenario in most cases (the only person allowed to fetch the transactions for a user id is that user, who owns the transactions).
But owner is definitely a bit cryptic.

Reply:
ohhh it's owning the _transactions_ not the _user_! ok I didn't catch that 😅 - I think that's hard to infer since the context is removed from what you're owning